glossary.yml

```
@@ -1856,11 +1870,21 @@
     def: >
       A two-dimensional data structure for storing tabular data in memory. Rows
       represent [records](#record) and columns represent [variables](#variable_data).
+      Each column in a data frame is an element of a list with the same length
+      so each column can have a different [type](#data_type).
```

```suggestion
      Every column in a data frame has the same length, and contains one type of 
      [data](#data_type), allowing each column to have a different type.
```

```suggestion
      A data frame works like a list of columns; this allows each column to hold a
      different type of [data](#data_type). However, data frames are differentiated
      from more general lists of lists by requiring that each column have the same length.
```

I find the 'is an element of a list' no super clear; I see what you mean, but I think the wording could confuse people who are just learning about data frames because they may not have reached a point where they think about data structures in terms of other data structures.

I've offered two different ways this could be reworded that I think avoid this because I'm not sure which aspects of the definition you consider more important.

---

Thanks, I think your first one is more concise for a wider audience. I worked with data frames for a long time before learning they were a special type of lists. It explained how they work to me but may confuse beginners.